### PR TITLE
build: fixed type checking.

### LINKED ISSUE #343: Fix `typecheck` pre-commit hook
We already have `lint-staged` hook, but it does not work.

`typecheck` pre-commit hook is important for ts codebase

- reduce bugs: a couple of previous issues I submitted are related to type errors.
- we can avoid building errors on PRs, reduced time and machine resources.

It seems that our typescript version is too low (current: `4.0.8`, latest: `4.5.5`), maybe eslint config upgrade is also required, and husky is not configured, so It even doesn't run when using `git commit`.

Another work to do is to fix existing type errors in the codebase.:
type errors

### PR DESCRIPTION
**Pull request by zhyd1997**
resolved #343 
adr decisions

### CODE REVIEW

File: src/store/main/sagas/loadReadOnlyData.ts [hunk 1]
```diff
@@ -15,7 +15,7 @@ export function* loadReadOnlyData() {
       coingeckoPrices,
       isLoading: false,
     }));
-  } catch (e) {
+  } catch (e: any) {
```
Comment (bigint) [inline src/store/main/sagas/loadReadOnlyData.ts:18]: Can we use `error` type instead of `any` 🙇🏻

**Review by bigint — CHANGES_REQUESTED** [review]

File: src/store/main/sagas/loadReadOnlyData.ts (hunk 1 above)
Comment (zhyd1997) [reply src/store/main/sagas/loadReadOnlyData.ts]: Done 0357287

**Review by bigint — APPROVED** [review]
LGTM 🚀

**Comment by zhyd1997** [comment]
@samirsalem How to test CI locally

**Review by samirsalem — COMMENTED** [review]
reviewed this, can be merged

**Review by samirsalem — APPROVED** [review]

**Pull request merged**